services/iiif/image.js

```diff
@@ -0,0 +1,40 @@
+const axios = require("axios");
+
+const getService = async (service, preferredWidth=1000) => {
+  if (service.hasOwnProperty('@id')){
+    try {
+      const response = await axios.get(`${service['@id']}/info.json`);
+      const {width, height} = response.data.sizes.reduce((a, b)=> {
+        return Math.abs(b.width - preferredWidth) < Math.abs(a.width - preferredWidth) ? b : a;
+      });
+      return {
+        'id': `${service['@id']}/full/${width},${height}/0/default.jpg`,
+        'width': width,
+        'height': height,
+      }
+    }
+    catch (err) {
+      console.log(err.response.status);
+      return {}
+    }
+  }
+};
+
+exports.getRepresentativeImage = async (resource, preferredSize=1000) => {
+  const firstCanvas = resource.items[0].items[0].items[0];
+  const firstCanvasService = await getService(firstCanvas.body.service[0], preferredSize);
+  if (firstCanvasService !== {}) {
+    return [
+      {
+        "id": firstCanvasService.id,
+        "type": "Image",
+        "format": "image/jpeg",
+        "width": firstCanvasService.width,
+        "height": firstCanvasService.height,
+      }
+    ]
+  }
```

Comment:
At some point, we'll need to get a little more defensive  in here for various Canvas cases to make this more usable outside of the call we have. I think we should push it through for now as it works with this example collection and its manifests -- then build some unit tests around various canvas shapes... I know we don't have really any testing at this point, but maybe this would be a good situation to work on that.

- Line 25 assumes a service exists, we may just want to check if that exists before calling the `getService()`
- Just to be extra defensive we might want to check if resource.items (manifest.items) exists and that has at least one canvas. It MUST according to https://iiif.io/api/presentation/3.0/#items but I have seen cases in the wild where that isn't true and we should probably try to get Canopy to proceed without bombing if it comes across this case.